extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/TaskAdapter.java

```diff
@@ -31,4 +32,5 @@ public interface TaskAdapter
 
   Task toTask(Job from) throws IOException;
 
+  K8sTaskId getTaskId(Job from) throws IOException;
```

Comment:
Seems this method has same implementation in the subclasses, maybe move here as a default method.

Reply:
i didn't want to do this because the implementation depends on logic in fromTask, since both adapters happen to add the same annotation for taskId the implementation for getTaskId is the same but this isn't necessarily the case